### PR TITLE
Specify arguments using JSON configuration file, provide welcome page as argument/option

### PR DESCRIPTION
**Pull request by cdarby**
```
x <- list(
  Azimuth.app.reference = "/Users/charlotte/Documents/nygc_lab/reference_mapping/pbmc_i21",
  Azimuth.app.demodataset = "/Users/charlotte/Documents/nygc_lab/azimuth_testdata/pbmc_10k_v3_filtered_feature_bc_matrix.h5",
  Azimuth.app.plotseed = 28,
  Azimuth.map.pbcorthresh = 0.5
)
jsonlite::write_json(x = x, path = 'tmp.json', auto_unbox=TRUE, pretty=TRUE)

Azimuth::AzimuthApp(config = "tmp.json", reference = "NONE")
```

- [ ] Arguments specified in the command are currently overwritten by arguments specified in the config file. Any suggestions for a different conflict-resolving strategy?

### CODE REVIEW

**Comment by cdarby** [comment]
Welcome box is blank if the option Azimuth.app.welcomebox or AzimuthApp argument welcomebox is not specified. Here is the code to generate the box for the PBMC app:

```
welcomeboxpbmc <- 
  "box(
    h3(\"Please upload a dataset to map to the Multimodal PBMC reference\"),
    \"Upload a counts matrix from an scRNA-seq dataset of human PBMC in one
        of the following formats: hdf5, rds, h5ad, h5seurat. For testing, we
        also provide a demo dataset of 11,769 human PBMC from 10x Genomics,
        which is loaded automatically with the 'Load demo dataset' button
        or available for download \",
    a(\"here\",
      href=\"https://www.dropbox.com/s/cmbvq2og93lnl9z/pbmc_10k_v3_filtered_feature_bc_matrix.h5?dl=0\",
      target=\"_blank\"), # open in new browser tab
    \".\",
    width = 12
  )"

Azimuth::AzimuthApp(reference = "/Users/charlotte/Documents/nygc_lab/reference_mapping/pbmc_i21", welcomebox = welcomeboxpbmc)
```

- [ ] We could actually include the PBMC config file (and later other config files) in the repo. That will save us from having to upload this file separately to AWS S3 and pull it on the workers, we'd just give the path with respect to the repository source files in the docker image. Thoughts?

**Comment by andrewwbutler** [comment]
1. Would it make more sense to have command line arguments override the config file instead? Or perhaps just error out if there are conflicts?
2. I think including the PBMC config file is a good idea so people at least have an example. Not sure if we'd want to include later ones but can decide that when we release other references.

File: inst/resources/config.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  "Azimuth.de.digits": 3,
+  "Azimuth.de.mincells": 15,
+  "Azimuth.map.ncells": 100,
+  "Azimuth.map.ngenes": 250,
+  "Azimuth.map.nanchors": 50,
+  "Azimuth.map.ntrees": 20,
+  "Azimuth.map.pbcorthresh": 0.75,
+  "Azimuth.sct.ncells": 2000,
+  "Azimuth.sct.nfeats": 2000
+}
```
[the remark this answers is not in the document]
Comment (cdarby) [reply inst/resources/config.json:19]: @mojaveazure Is something like this more helpful, showing both ways to write Azimuth.app options, and examples of other options? (I'm assuming that this file would not be read in by default in the config parameter to AzimuthApp.)
```
{
  "Azimuth.app.reference": "/path/to/reference",
  "welcomebox": "box(h3(\"Header\"), \"body text\", a(\"link\", href=\"www.satijalab.org\", target=\"_blank\"), width = 12)"
  "Azimuth.map.pbcorthresh": 0.5,
  "another.package.option": 100,
}
```

**Review by mojaveazure — APPROVED** [review]
This looks good. I've made some additional changes:
- All dotted options in the `Azimuth.app` namespace have been converted to use underscores instead. I've updated the documentation, the code, and the README to reflect these changes
- I've slightly reorganized the documentation of package options in `?Azimuth`
- The example configuration file has been updated to be a realistic example, including setting options found in external packages. Code used to generate the example config file has been provided as well
- I've made some changes to how certain external options get resolved (namely `shiny.maxRequestSize`, `DT.options`, and `future.globals.maxSize`)
- I've added some notes about shorthand-notation for `Azimuth.app` options in the documentation for `AzimuthApp`

**Comment by mojaveazure** [comment]
Finally, I've bumped the version to 0.2.0 as this represents a significant change in functionality with regards to launching the app

**Pull request merged**